src/02-room-details.js

```diff
@@ -49,7 +67,28 @@ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
       "Kit Hopkins Education Wing"
     ]
  */
-function getConnectedRoomNamesById(rooms, id) {}
+function getConnectedRoomNamesById(rooms, id) {
+  const matchingRoom = rooms.find(room => room.roomId === id);
+
+ 
+      if(!matchingRoom){
+    return `Room with ID of 'incorrect-id' could not be found.`;
+  }
+
+    if (matchingRoom) {
+    if (matchingRoom.connectsTo.length === 0) {
+      return `Room '${matchingRoom.name}' has no connected rooms`;
+    } else {
+      const connectedRooms = matchingRoom.connectsTo.map(connId => {
+        const connectedRoom = rooms.find(room => room.roomId === connId);
```

Comment:
I don't think we need a map here  - why not just grab the id and run the find?

Slow down - space this out step by step and don't worry about writing out in a fancy way